documentation/specs/src/base-ledger/governance.md

```diff
@@ -108,81 +122,97 @@ This means that corresponding VPs need to handle these cases.
 
 ### Proposal Transactions
 
-The proposal transaction will have the following structure, where `author` address will be the refund address.
+The on-chain proposal transaction will have the following structure, where `author` address will be the refund address.
 
 ```rust
-struct OnChainProposal {
-    id: u64
-    content: Vec<u8>
-    author: Address
-    votingStartEpoch: Epoch
-    votingEndEpoch: Epoch
-    graceEpoch: Epoch
-    proposalCode: Option<Vec<u8>>
+struct Proposal {
+    id: u64,
+    content: Vec<u8>,
+    author: Address,
+    proposal_type: ProposalType,
```

Comment:
toCamelCase()

Reply:
To use `type` as the name of the field we actually need to use the raw identifier `r#type` since `type` is a reserved keyword. Is it ok?